src/apps/components/AppFrame/useAppActions.ts

```diff
@@ -40,8 +48,24 @@ export const useAppActions = (
     action: Actions | undefined,
   ): DispatchResponseEvent => {
     switch (action?.type) {
+      case "notification": {
+        const {
+          actionId,
+          ...notification
+        } = action.payload as NotificationAction["payload"];
+
+        notify({
+          ...notification,
+        });
```

Comment:
Can we add some metadata like small text "from: XYZ app"?

Yes, this PR introduces great "native like" feeling, but it also exposes some attack vector to mimic native messages from Core/Dashboard.

Also, it can be debugged easier, to see the source 

cc @timuric

Reply:
Yes, that sounds like a nice addition. I think we would have to modify the UI of notification a bit to make it look nicer. Maybe let's do it in another PR?

Reply:
Please talk to @pwgryglak about this and let us know if he wants to make something fancy or just add one line of text somewhere. We can do the latter right now in this PR

Reply:
@dominik-zeglen @Cloud11PL 

if this requires some bigger changes and design process, we can have this in another PR

Reply:
It depends really on the proposed design. Let's wait for @pwgryglak input :)

Reply:
This will be done in a separate PR as per this issue - https://github.com/saleor/macaw-ui/issues/158